### PR TITLE
soc: stm32f1: add port uart4

### PR DESCRIPTION
**Pull request by jli157**
Enable supporting UART4 on STM32F103Xe and STM32F107 SoCs.

Signed-off-by: Jun Li <jun.r.li@intel.com>

### CODE REVIEW

File: dts/arm/st/stm32f1.dtsi [hunk 1]
```diff
@@ -123,6 +123,15 @@
 			label = "UART_3";
 		};
 
+		usart4: serial@40004c00 {
```
Comment (erwango) [inline dts/arm/st/stm32f1.dtsi]: Looking in ext/hal/st/stm32cube/stm32f1xx/soc/stm32f1*.h files, it appears UART4 is present only in half of SoCs of STM32F1 series.
So, if you need UART4, you need to add it in the variant dsti file (stm32f103xe.dsti or stm32f107.dtsi)

**Review by erwango — CHANGES_REQUESTED** [review]
One comment, otherwise looks good

**Comment by ydamigos** [comment]
In ext/hal/st/stm32cube/stm32f1xx/soc/stm32f1*.h files, it appears UART4 and in this PR all it appears as USART4, shouldn't we change it to UART4 like in the DT files of stm32f4 series?

**Comment by erwango** [comment]
> In ext/hal/st/stm32cube/stm32f1xx/soc/stm32f1*.h files, it appears UART4 and in this PR it appears as USART4, shouldn't we change it to UART4 like in the DT files of stm32f4 series?

Indeed, this is UART and not USART, thanks for catching that.

File: dts/arm/st/stm32f107.dtsi [hunk 2]
```diff
@@ -5,3 +5,17 @@
  */
 
 #include <st/stm32f1.dtsi>
+
+/ {
+	soc {
+
```
Comment (erwango) [inline dts/arm/st/stm32f107.dtsi]: remove this blank line

File: dts/arm/st/stm32f103Xe.dtsi [hunk 3]
```diff
@@ -11,6 +11,15 @@
 
 / {
 	soc {
+		uart4: serial@40004c00 {
+			compatible = "st,stm32-usart", "st,stm32-uart";
```
Comment (erwango) [inline dts/arm/st/stm32f103Xe.dtsi]: Remove compatible "st,stm32-usart"

File: dts/arm/st/stm32f107.dtsi [hunk 4]
```diff
@@ -5,3 +5,17 @@
...
+		uart4: serial@40004c00 {
+			compatible = "st,stm32-usart", "st,stm32-uart";
```
Comment (erwango) [inline dts/arm/st/stm32f107.dtsi]: remove compatible "st,stm32-usart"

**Review by erwango — CHANGES_REQUESTED** [review]
Some additional comments.
Otherwise we're good

**Comment by jli157** [comment]
Revised the code again based on the further comments. Also, fixed naming bugs in the file: arch/arm/soc/st_stm32/stm32f1/dts.fixup where all `ST_STM32_USART_40004C00` shall be replaced by `ST_STM32_UART_40004C00`.

**Review by ydamigos — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**